user/src/com/google/gwt/user/client/ui/RootPanel.java

```diff
@@ -179,6 +179,12 @@ public static RootPanel get(String id) {
       // for any reason, return a new RootPanel rather than the existing one (
       // see issue 1937).
       if ((elem == null) || (rp.getElement() == elem)) {
+        // if id and elem are null, set the element of existing rp with the 
+        // current body to handle scenarios where body gets replaced/refreshed
+        // from DOM on certain actions like page navigation.
+        if(id == null && elem == null) {
```

Comment:
If `id` is null, then `elem` is necessarily null as well (`elem` is only assigned at line 172 when `id` is not null); and conversely, if `elem` is null at that point then `id` is null (because at lines 172–173, when `id` is not null and `elem` is null, then the method returns).

Reply:
Resolved , placed a single (id == null) check only